moegwsa/accurev-scm-plugin
# Dev (#20)

* made integration test run with SCM-head events and added new trigger files

* renamed triggers/jenkinsConfig.JSON to triggers/jenkinsConfig.json

* use image from dockerhub

* switching docker-hub repository for dev stream

* fixed stream filters again, and updated automated integration tests

* trigger new build

* point to right docker image

* enabled ignored test, should run a total of 87 tests now

* skip test false

* tests fail on warnings

* try reading test-reports

* enable stacktrace

* full stacktrace enabled

* skip equal to false

* always read test output

* hide empty gating streams

* changed to not look at last build, but current transaction when computing changelog for staging streams

* Feature/default transactions (#12)

* added new accurev client version, and changed to use default CAt from default group instead of latest

* add mvn install to before_install in order to read settings.xnl

* changed the install command to use default with settings.xml file as input

* register api_token in netrc

* rename settings.xml to  .travis.settings.xml

* cp .travis.settings.xml $HOME/.m2/settings.xml

* Feature/cleanup triggerfiles (#13)

* reenabled admin requirement to better reflect original example triggers from accurev

* First review with example files

* moved common functions to Utils, cleanup webhook to not check for $command, renamed to copyinputfile to cacheinputfile

* changed to use constants instead of parseToReason method in webhook

* cAT for staging streams is now set to a single transaction, and mqqt-responce is sent by first build data incountered

* added dependency to strict-crumb issuer, changed where temp cache file is found, add events

* added https to jenkins.config instead of appending in trigger files

* filter gated stream

* Feature/changelog summary (#14)

* reenabled admin requirement to better reflect original example triggers from accurev

* First review with example files

* moved common functions to Utils, cleanup webhook to not check for $command, renamed to copyinputfile to cacheinputfile

* changed to use constants instead of parseToReason method in webhook

* cAT for staging streams is now set to a single transaction, and mqqt-responce is sent by first build data incountered

* added dependency to strict-crumb issuer, changed where temp cache file is found, add events

* added https to jenkins.config instead of appending in trigger files

* build changes no longer link to it self, a summary shows up in recent changes for job

* nochanges does not always show up anymore

* implemented repository browser, still needs ui configuration

* changed the GUI for changelog and the link to changeset, removed difflink

* display warnings, if jenkins isn't available (#16)

* calculate simple affected files from string

* backup new trigger files

* generate warning for unreachable server

* Bug/multibranch pipeline events (#17)

* headevent now uses own retrieve method

* filter of enabled webhooks

* use ^ for enable webhook on all streams

* Filter events by eventype and stream name in jobs

* Added upper bound for calculating accurev commands when recieving for head event.

* Updated event flow to use incomming payload stream and transaction, when determining what to build.

* Feature/staging trait (#19)

* Added build discovery trait for staging streams

* Fixed capitalization of staging stream trait. Removed if statement causing double changelog calculation in determineTransToBuild

* Revoked removal of second if candidates, and moved it into first if candidates, due to null pointer in freestyle projects.

* Collected getCandidateTransaction to a private method, to avoid dublicated code, manual builds now usage timestamp now to calculate changelog, nullcheck for issues in broker on autopromote.

* syntax error in broker, us defined instead of undef for issue check

* updated versionnumber
